services/harmony/app/frontends/jobs.ts

```diff
@@ -134,6 +135,9 @@ export async function getJobsListing(
     let listing;
     await db.transaction(async (tx) => {
       listing = await Job.queryAll(tx, query, page, limit);
+      for (const job of listing.data) {
```

Comment:
I think it'd be nice to have this as an optional return field from queryAll so that we can reuse it (for the workflow ui for example). Also wondering why you decided to use a loop versus doing a join between jobs and labels?

Reply:
I started to add this to `queryAll`, but it would mean adding another parameter to the function call, which already has a lot of optional parameters. I didn't want to deal with having to update the call in all the places it's used. And I wasn't sure if labels would be useful in most places where jobs are pulled from the dB, so I would be adding a parameter just to accommodate a handful of uses. But I could be persuaded otherwise. 

A join would only make sense when retrieving the jobs, so I would have to add the join to the `knex` `select` happening in `queryAll` - may be worth trying. I'm kind of growing weary of the places where we build these queries and optionally add additional `knex` actions (like `join`) based on the function parameters. But maybe that's just me. Would like to hear from others.

Reply:
Alternatively, I could just query for the labels in bulk using the job IDs, instead of iterating over the jobs.

Reply:
I think we allow up to 2000 jobs so we definitely should remove the `for` loop querying one job at a time. I don't really have a strong opinion on whether using `includeLabels` as a parameter vs. a second query to grab the labels when you need them is better. I guess the pros are what Vinny mentioned about the worklow-ui just having to add `includeLabels=true` as well as only issuing one query.

Reply:
> I started to add this to queryAll...
Now that I think about it some more, I may want to query for them separately anyway so that the UI loads faster--but either way it's ok.

Reply:
I changed it to use a double-join to (optionally) get the labels in `queryAll`.